components/brave_vpn/common/mojom/brave_vpn.mojom

```diff
@@ -73,6 +73,14 @@ interface ServiceHandler {
   [EnableIfNot=is_android]
   ResetConnectionState();
 
+  [EnableIfNot=is_android]
+  EnableOnDemand(bool enable);
```

Comment:
It could be just be me, I'm a little bit confused if it's `!onDemand == autoReconnect` or backward. Could you add a comment here?

Reply:
`on demand` and `automatic reconnect` both have same meaning.
I think `automatic reconnect` is more easily explained terms about `on demand`.
When `on demand` is enabled, vpn is connected(`automatically`) when network request is made on the platform. That means vpn is connected on demand only when network request is made.
I'll add some more comments.

Reply:
Thanks for the explanation 👍